data/campaigns/Winds_of_Fate/units/Gates.cfg

```diff
@@ -0,0 +1,64 @@
+#textdomain wesnoth-wof
+
+[movetype]
+    name=gate
+    [movement_costs]
+        deep_water=1
+        shallow_water=1
+        reef=1
+        swamp_water=1
+        flat=1
+        sand=1
+        forest=1
+        hills=1
+        mountains=1
+        village=1
+        castle=1
+        cave=1
+        frozen=1
+        unwalkable=1
+        fungus=1
+    [/movement_costs]
+    [resistance]
+        blade=50
+        pierce=50
+        impact=100
+        fire=60
+        cold=60
+        arcane=40
+    [/resistance]
+[/movetype]
+
+[unit_type]
+    id=Gate
+    name= _ "Gate"
+    race=mechanical
+    image="data/core/images/scenery/gate-rusty-sw.png"
+    profile="data/core/images/scenery/gate-rusty-sw.png"
+    hitpoints=80
+    movement_type=gate
+    movement=0
+    experience=25
+    level=0
+    alignment=neutral
+    advances_to=null
```

Comment:
You could probably add `hide_help=yes` and `do_not_list=yes` somewhere so this pseudo-unit isn't listed in help or website.

Reply:
Makes sense; added these settings.

Reply:
Can't hidden unit help files still be seen in when accessed through a unit on map?  I believe it just gets hidden when browsing the help.

Reply:
> Can't hidden unit help files still be seen in when accessed through a unit on map? I believe it just gets hidden when browsing the help.
Yes. Though why would we want to prevent this unit from showing up in all possible pathways the player might use to look for it?

Reply:
Unless the gate is like the broom from _The Scorcerer's Apprentice_, the "unit" is a hack, and adding it to the bestiary makes that even more noticeable.  Is the info that important that the player needs it when the gates are no longer present?

Reply:
Those resistances _can_ be seen when hovering over the gate, right? So it doesn't necessarily need to be available in help?

Reply:
@doofus-01
> Unless the gate is like the broom from _The Scorcerer's Apprentice_, the "unit" is a hack, and adding it to the bestiary makes that even more noticeable.

After some testing, it looks like the gate unit only shows up in the help menu while you are playing a WoF scenario. If you are playing any other campaign, or multiplayer, none of the custom WoF-specific units will be visible in the help menu.

This may be a new feature of 1.15, that campaign specific units are only listed while you are playing that specific campaign.

> Is the info that important that the player needs it when the gates are no longer present?

The issue would be more that a newer player may not yet know to use the keyboard shortcut "d" while on a unit, or to right-click on a unit and select "unit description" (which only appears in the right click menu if the click was directly on a unit, so the discoverability of this method is not the best).

If a newer player only knows to navigate the help menu to find a unit's resistances, then they would be out of luck, if we apply hide_help to the Gates.

@CelticMinstrel 
> Those resistances _can_ be seen when hovering over the gate, right?

Surprisingly, no.
Only a unit's present terrain defense is shown. You would think the resistances would show up on the sidebar when you hover or select a unit, but this information is hidden away in the unit description.

Reply:
> This may be a new feature of 1.15, that campaign specific units are only listed while you are playing that specific campaign.

I don't think that's the case at all.

> Surprisingly, no.
> Only a unit's present terrain defense is shown. You would think the resistances would show up on the sidebar when you hover or select a unit, but this information is hidden away in the unit description.

Ah, I was wrong. Click the gate, then hover over the HP in the sidebar.

In other words, the information _is_ available even without the help. It's perhaps a bit unfortunate that it's not visible when just hovering the unit, though.

Reply:
> I don't think that's the case at all.

It is easy enough to test for yourself.

Pick out an easy to find custom unit from any campaign, for example _The Sceptre of Fire_ has a [Mechanical "Caravan" unit that appears in its first scenario.](https://units.wesnoth.org/1.15/mainline/en_US/Sceptre_of_Fire.html) Then load a multiplayer scenario or any other campaign, use the "debug" command so the help menu is populated with all available units, and see if you see that same Caravan unit listed under the "Mechanical" category.

Likewise, if you have _Winds of Fate_ installed, see if you see the Gate unit in the help menu from any scenario or campaign other than _Winds of Fate_.

> Ah, I was wrong. Click the gate, then hover over the HP in the sidebar.

I have played wesnoth since 2005 and I had no idea this method was available. That is the trouble with relying on these obscure shortcuts; many newer players will not even know they are there due to their low discoverability.

Reply:
> It is easy enough to test for yourself.

I mean, I don't think that's an engine change. The create unit command just shows all units that are accessible, right? So if SoF included its unit definitions outside of the campaign define, then they'd be visible in MP.

Although it _is_ possible that the change to load only active add-ons affects this in some way too. That _is_ new in 1.15.x.

Reply:
Sounds quite plausible. It might also be I only _noticed_ this feature in 1.15.x but it was originally implemented in 1.13.x or 1.11.x.